packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/handler.ts

```diff
@@ -0,0 +1,222 @@
+
+    try {
+      const resp = await this.eks.describeCluster({ name: this.clusterName });
+      console.log('describeCluster returned:', JSON.stringify(resp, undefined, 2));
+    } catch (e) {
+      if (e.code === 'ResourceNotFoundException') {
+        console.log('received ResourceNotFoundException, this means the cluster has been deleted (or never existed)');
+        return { IsComplete: true };
+      }
+
+      console.log('describeCluster error:', e);
+      throw e;
+    }
+
+    return {
+      IsComplete: false
+    };
+  }
+
+  // ------
+  // UPDATE
+  // ------
+
+  public async onUpdate() {
+    const updates = analyzeUpdate(this.oldProps, this.newProps);
+
+    console.log(updates);
+
+    // if there is an update that requires replacement, go ahead and just create
+    // a new cluster with the new config. The old cluster will automatically be
+    // deleted by cloudformation upon success.
+    if (updates.replaceName || updates.replaceRole || updates.replaceVpc) {
+      return await this.onCreate();
+    }
+
+    // if a version update is required, issue the version update
+    if (updates.updateVersion) {
+      if (!this.newProps.version) {
+        throw new Error(`Cannot remove cluster version configuration. Current version is ${this.oldProps.version}`);
+      }
+
+      await this.updateClusterVersion(this.newProps.version);
+    }
+
+    if (updates.updateLogging || updates.updateAccess) {
+      return await this.eks.updateClusterConfig({
+        name: this.clusterName,
+        logging: this.newProps.logging,
+        resourcesVpcConfig: this.newProps.resourcesVpcConfig
+      });
+    }
+
+    // no updates
+    return;
+  }
+
+  public async isUpdateComplete() {
+    return this.isActive();
+  }
+
+  private async updateClusterVersion(newVersion: string) {
+    // update-cluster-version will fail if we try to update to the same version,
+    // so skip in this case.
+    const cluster = (await this.eks.describeCluster({ name: this.clusterName })).cluster!;
+    if (cluster.version === newVersion) {
+      console.log(`cluster already at version ${cluster.version}, skipping version update`);
+      return;
+    }
+
+    await this.eks.updateClusterVersion({ name: this.clusterName, version: newVersion });
+  }
+
+  private async isActive(): Promise<IsCompleteResponse> {
+    console.log('waiting for cluster to become ACTIVE');
+    const resp = await this.eks.describeCluster({ name: this.clusterName });
+    console.log('describeCluster result:', JSON.stringify(resp, undefined, 2));
+    const cluster = resp.cluster!;
+    if (cluster.status !== 'ACTIVE') {
+      return { IsComplete: false };
+    }
+
+    return {
+      IsComplete: true,
+      Data: {
+        Name: cluster.name,
+        Endpoint: cluster.endpoint,
+        Arn: cluster.arn,
+        CertificateAuthorityData: cluster.certificateAuthority && cluster.certificateAuthority.data
+      }
+    };
```

Comment:
The resource provider framework explicitly forbids returning data as long as the operation is not complete (otherwise, it will need to somehow synthesize the state from all retries).

`Data` is the CloudFormation resource attributes (`Fn::GetAtt`). `Status` is not an attribute of the resource since it's a transient value.

Reply:
Oh I see. Thanks for RTFM for me :)